tsl/src/continuous_aggs/Readme.md

```diff
@@ -1,144 +1,108 @@
+to refresh a subset of the continuous aggregate at relatively low
+cost, thus enabling continuous aggregation as new data is written or
+old data is updated and/or backfilled.
+
+To enable continuous aggregation, a continuous aggregate stores
+partial aggregations for every time bucket in an internal
+hypertable. This means that each time bucket can be recomputed
+individually, without requiring updates to other buckets. Finalization
+of the partial buckets happens automatically at query time. Although
+such finalization gives slightly higher querying times, it is offset
+by much more efficient refreshes that only recompute the buckets that
+have been "invalidated" by changes in the raw data.
+
+A continuous aggregate policy automates the refreshing, allowing the
+aggregate to stay up-to-date without manual intervention. A policy can
+be configured to only refresh the most recent data (e.g., just the
+last hour's worth of data) or ensure that the continuous aggregate is
+always up-to-date with the underlying source data. Policies that focus
+on recent data allow older parts of the continuous aggregate to stay
+the same or be governed by manual refreshes.
+
+## Bookkeeping and Internal State ##
+
+TimescaleDB does bookkeeping for each continuous aggregate to know
+which buckets of the aggregates require refreshing. Whenever a
+modification happens to the source data, an invalidation for the
+modified region is written to an invalidation log. However,
+invalidations are not written beyond the *invalidation threshold*,
+which tracks the largest bucket materialized thus far. This threshold
+allows write amplification to be kept to a minimum by, e.g, never
+refreshing the most recent time bucket that is still being "filled".
+
+Thus, to store, maintain, and query aggregations, continuous
+aggregates consists for the following objects and state:
+
+1. A query view, which finalizes the aggregations and is also the
+   object that users interact with.
+2. A partial view, which is used to materialize new data.
+3. A direct view, which holds the original query that users specified.
+4. An internal materialization hypertable, containing the materialized
+   data for each time bucket.
+5. An invalidation threshold, which tracks how "far" the
+   materialization has reached and below which invalidations must be
+   logged.
+6. An invalidation log. Entries in this log contain time ranges that
+   should be re-materialized during refreshes, as an INSERT, UPDATE,
+   DELETE, or other mutating operation may have altered the source
+   data. In practice, this log is split in two; once a refresh is run
+   on a continuous aggregate, each new invalidation is copied to a
+   per-materialization log that tracks which invalidations have been
+   processed for each continuous aggregate. This is necessary since
+   there might be multiple continuous aggregate per source hypertable,
+   and each of them might be refreshed and materialized separately.
```

Comment:
Maybe describe the two logs as two separate objects in that case. I think it would also make the description easier to follow.